plugins/extra/karma.go

```diff
@@ -5,35 +5,93 @@ import (
 	"strings"
 	"unicode"
 
+	"github.com/go-xorm/xorm"
+
 	"github.com/belak/go-seabird"
-	"github.com/belak/nut"
 	irc "github.com/go-irc/irc/v2"
+
+	"github.com/belak/nut"
 )
 
 func init() {
 	seabird.RegisterPlugin("karma", newKarmaPlugin)
 }
 
 type karmaPlugin struct {
-	db *nut.DB
+	db *xorm.Engine
 }
 
-// KarmaTarget represents an item with a karma count
-type KarmaTarget struct {
-	Name  string
+// Karma represents an item with a karma count
+type Karma struct {
+	ID    int64
+	Name  string `xorm:"unique"`
 	Score int
 }
 
 var regex = regexp.MustCompile(`([\w]{2,}|".+?")(\+\++|--+)(?:\s|$)`)
 
-func newKarmaPlugin(b *seabird.Bot, m *seabird.BasicMux, cm *seabird.CommandMux, db *nut.DB) error {
+func newKarmaPlugin(b *seabird.Bot, m *seabird.BasicMux, cm *seabird.CommandMux, ndb *nut.DB, db *xorm.Engine) error {
+	l := b.GetLogger()
 	p := &karmaPlugin{db: db}
 
-	err := p.db.EnsureBucket("karma")
+	// Migrate any relevant tables
+	err := db.Sync(Karma{})
 	if err != nil {
 		return err
 	}
 
+	rowCount, err := db.Count(Karma{})
+	if err != nil {
+		return err
+	}
+
+	// If a nut DB exists, we need to migrate all the data
+	if ndb != nil && rowCount != 0 {
+		l.Info("Skipping karma migration because target table is non-empty")
```

Comment:
Add return here and outdent the rest of the method

Reply:
I guess that would be possible... I'd have to move the event registration to before migration though.

Reply:
This was moved into a separate command so the plugin file is much cleaner now